examples/ex9.cpp

```diff
@@ -263,11 +282,14 @@ int main(int argc, char *argv[])
       new BoundaryFlowIntegrator(inflow, velocity, -1.0, -0.5));
 
    m.Assemble();
-   m.Finalize();
    int skip_zeros = 0;
    k.Assemble(skip_zeros);
-   k.Finalize(skip_zeros);
    b.Assemble();
+   if (!pa)
+   {
+      m.Finalize();
+      k.Finalize(skip_zeros);
+   }
```

Comment:
I'd move the check for assembly level inside the `Finalize()` method -- keeps it simple in the example.